git/refs/symbolic.py

```diff
@@ -171,7 +172,13 @@ def _get_ref_info_helper(
         tokens: Union[None, List[str], Tuple[str, str]] = None
         repodir = _git_dir(repo, ref_path)
         try:
-            with open(os.path.join(repodir, str(ref_path)), "rt", encoding="UTF-8") as fp:
+            # Make path absolute, resolving any symlinks, and check that we are still
+            # inside the repository
+            full_ref_path = Path(repodir, str(ref_path)).resolve()
```

Comment:
I think this will slow down each read of a loose ref, and maybe that can be avoided?
References mentioned in a symbolic ref are never absolute, and can be rejected on that ground. Then, when determined relative, one should normalize the `../` away to check if it would break out. So `refs/../foo` would be fine, but `refs/../foo/../../bar` would not be. All this can happen without resolving symlinks.

Reply:
I replaced the calls to `resolve()` with `os.path.abspath`, which normalizes `..` but does not resolve symlinks.